.github/workflows/publish_snapshot.yml

```diff
@@ -0,0 +1,102 @@
+name: Publish snapshots to maven
+
+on: [push, pull_request]
```

Comment:
This should be only push. The pull request should not be able to trigger this work. Also specify the branch maybe?